lib/libc/gen/psignal.c

```diff
@@ -61,3 +62,91 @@ psiginfo(const siginfo_t *si, const char *s)
 {
 	psignal(si->si_signo, s);
 }
+
+int
+sig2str(int signum, char *str)
+{
+	char tmp[16];
+	char *t, *p;
+	int n;
+
+	if (signum > 0 && signum < sys_nsig) {
+		strcpy(str, sys_signame[signum]);
+		return (0);
+	}
+
+	if (signum < SIGRTMIN || signum > SIGRTMAX)
+		return (-1);
+
+	if (signum <= (SIGRTMIN + SIGRTMAX) / 2) {
+		strcpy(str, "RTMIN");
+		n = signum - SIGRTMIN;
+	} else {
+		strcpy(str, "RTMAX");
+		n = signum - SIGRTMAX;
+	}
+
+	if (n != 0) {
+		/*
+		 * This block does the equivalent of
+		 * sprintf(str + 5, "%+d", n);
```

Comment:
Then why cannot the code use s(n)printf() directly, and needs to hand-roll the printf?

Reply:
> Then why cannot the code use s(n)printf() directly, and needs to hand-roll the printf?

For the same reason it's done in strsignal.c:

> /* Do this by hand, so we don't link to stdio(3). */

https://github.com/freebsd/freebsd-src/commit/78121e79aa390b2e94fb8aca2d0477d81aa20cf2#diff-4615e09983d3d0fad76f04357ef5f26d3ee008e7f3f21e71dfd1a85035d04279R56

I can change it, of course.

Reply:
Fixed.